### PR TITLE
add edit_merge_request_comment

### PR DESCRIPTION
**Pull request by hjanuschka**
* add edit_merge_request_comment -> for editing a MR note 
 * port from comments to newer 'notes' -> to get back the id

### CODE REVIEW

File: lib/gitlab/request.rb [hunk 1]
```diff
@@ -49,6 +49,7 @@ def post(path, options={})
       validate self.class.post(@endpoint + path, options)
     end
 
+
```
Comment (NARKOZ) [inline lib/gitlab/request.rb]: :scissors:

**Comment by NARKOZ** [comment]
Thank you. Can you fix the tests?

**Comment by hjanuschka** [comment]
* master rebased
  * tests should pass now
  * empty line removed (as suggested)

hope you can :octopus:  this

File: lib/gitlab/request.rb [hunk 2]
```diff
@@ -48,7 +48,7 @@ def post(path, options={})
       set_authorization_header(options, path)
       validate self.class.post(@endpoint + path, options)
     end
-
+
```
Comment (NARKOZ) [inline lib/gitlab/request.rb]: :scissors:

**Comment by hjanuschka** [comment]
@NARKOZ damn i missed the last tab - fixed now - sorry for that.

**Comment by NARKOZ** [comment]
Thank you.

**Comment by RobinDaugherty** [comment]
@hjanuschka thanks for this change! Do you know if the delete action on this endpoint is available in Gitlab? I'm getting a 405 for deletion of a "comment" I created, and I can't find mention of a delete action in the gitlab source. PUT is working though.

**Comment by asedge** [comment]
I don't see any way to delete one either.

**Comment by hjanuschka** [comment]
@RobinDaugherty @asedge you both are right there seems to be now way to delete a MR-note via API.
i opened an issue on gitlab https://gitlab.com/gitlab-org/gitlab-ce/issues/14944.

btw. i have no clue why i added the delete - when there is no gitlab functionallity - anyway lets hope that gitlab does the change - and the gem automagically will work :).

**Comment by hjanuschka** [comment]
ok its in the gitlab-upstream 👏  - seems to be released in 8.7 or so!

https://gitlab.com/gitlab-org/gitlab-ce/issues/14944

**Pull request merged**